clusterloader2/run-e2e.sh

```diff
@@ -20,6 +20,8 @@ set -o pipefail
 
 CLUSTERLOADER_ROOT=$(dirname "${BASH_SOURCE[0]}")
 export KUBECONFIG="${KUBECONFIG:-${HOME}/.kube/config}"
+# If we are in kubemark (i.e. KUBEMARK_ROOT_KUBECONFIG is set), install CSI driver in this cluster.
+export CSI_DRIVER_KUBECONFIG=${KUBEMARK_ROOT_KUBECONFIG:-${KUBECONFIG}}
```

Comment:
This won't really work, because this has to be a real pod - and in kubemark cluster you only will run that pod on a hollow-pod (aka - it won't be running).

We need to do something similar to what we did with heapster:
https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/resources/heapster_template.json#L47

I.e. running it in a base cluster - and somehow connect it to the kubemark cluster

Reply:
We solved this problem for prometheus already: it runs in the "root" cluster (CL2 reads KUBEMARK_ROOT_KUBECONFIG and runs there).

It should work IMHO if we set env variables correctly.